java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/explorer/NodePropertiesProvider.java

```diff
@@ -0,0 +1,210 @@
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionParams;
+import org.netbeans.modules.java.lsp.server.protocol.CodeActionsProvider;
+import org.netbeans.modules.java.lsp.server.protocol.NbCodeLanguageClient;
+import org.netbeans.modules.parsing.api.ResultIterator;
+import org.openide.nodes.Node;
+import org.openide.util.Exceptions;
+import org.openide.util.Lookup;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ *
+ * @author Tomas Hurka
+ */
+@ServiceProvider(service = CodeActionsProvider.class)
+public class NodePropertiesProvider extends CodeActionsProvider {
+
+    private static final String COMMAND_GET_NODE_PROPERTIES = "java.node.properties.get";      // NOI18N
+    private static final String COMMAND_SET_NODE_PROPERTIES = "java.node.properties.set";      // NOI18N
+
+    private static final String PROP_NAME = "propName";      // NOI18N
+    private static final String PROP_DNAME = "propDispName";      // NOI18N
+    private static final String PROP_HTML_NAME = "propHtmlName";      // NOI18N
+    private static final String PROP_SHORT_NAME = "propShortName";      // NOI18N
+    private static final String PROP_PREF = "propPref";      // NOI18N
+    private static final String PROP_EXPERT = "propExpert";      // NOI18N
+    private static final String PROP_HIDDEN = "propHidden";      // NOI18N
+    private static final String PROP_CAN_READ = "propRead";      // NOI18N
+    private static final String PROP_CAN_WRITE = "propWrite";      // NOI18N
+    private static final String PROP_VAL_TYPE = "propType";      // NOI18N
+    private static final String PROP_VALUE = "propValue";      // NOI18N
+    private static final String PROPS = "props";      // NOI18N
+
+    private static final Set<String> COMMANDS = new HashSet<>(Arrays.asList(
+            COMMAND_GET_NODE_PROPERTIES, COMMAND_SET_NODE_PROPERTIES
+    ));
+
+    private final Gson gson = new Gson();
+
+    @Override
+    public List<CodeAction> getCodeActions(ResultIterator resultIterator, CodeActionParams params) throws Exception {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public CompletableFuture<Object> processCommand(NbCodeLanguageClient client, String command, List<Object> arguments) {
+        if (!COMMANDS.contains(command)) {
+            return CompletableFuture.completedFuture(null);
+        }
+        if (arguments == null || arguments.isEmpty()) {
+            return CompletableFuture.completedFuture(null);
+        }
+        TreeNodeRegistry r = Lookup.getDefault().lookup(TreeNodeRegistry.class);
+        if (r == null) {
+            return CompletableFuture.completedFuture(null);
```

Comment:
Not sure if 'tree service unavailable' is a valid state of the system; if not, throw an error.